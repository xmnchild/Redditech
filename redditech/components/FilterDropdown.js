import { View, Dimensions } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { useState } from 'react';
import DropDownPicker from 'react-native-dropdown-picker';

const windowWidth = Dimensions.get('window').width;

const FilterDropdown = ({ setFilter }) => {
    const [open, setOpen] = useState(false);
    const [value, setValue] = useState('new');
    const [items, setItems] = useState([
        { label: 'New', value: 'new', icon: () => <Icon name='rocket-sharp' /> },
        { label: 'Hot', value: 'hot', icon: () => <Icon name='flame-outline' /> },
        { label: 'Top', value: 'top', icon: () => <Icon name='ribbon-sharp' /> }
    ]);
    return (
        <View style={{ justifyContent: "center", flexDirection: 'column', alignItems: "center", alignContent: "center", backgroundColor: "#F2F2F2", height: 30, bottom: 10, zIndex: 1000 }}>
            <DropDownPicker
                open={open}
                value={value}
                items={items}
                placeholder={value}
                setOpen={setOpen}
                setValue={setValue}
                setItems={setItems}
                onChangeValue={(selectedValue) => {
                    setValue(selectedValue);
                    setFilter(selectedValue);
                }}
                style={{ backgroundColor: "#F2F2F2", borderWidth: 0, width: windowWidth, borderRadius: 0 }}
                dropDownContainerStyle={{ width: windowWidth, borderWidth: 0 }}
                textStyle={{ textTransform: 'uppercase', fontSize: 15, fontWeight: '500' }}
                showTickIcon={false}
                arrowIconContainerStyle={{ marginRight: 250 }}
            />
        </View>
    );
};


export default FilterDropdown;