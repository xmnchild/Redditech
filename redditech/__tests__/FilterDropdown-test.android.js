import * as React from "react";
import renderer from "react-test-renderer";
import FilterDropdown from '../components/FilterDropdown';

jest.mock('react-native-vector-icons');
jest.mock('react-native-dropdown-picker');
it(`renders correctly`, () => {
    const tree = renderer.create(<FilterDropdown/>);
    expect(tree).toMatchSnapshot();
});
