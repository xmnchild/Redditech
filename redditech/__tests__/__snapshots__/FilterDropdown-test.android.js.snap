// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "bottom": 10,
      "flexDirection": "column",
      "height": 30,
      "justifyContent": "center",
      "zIndex": 1000,
    }
  }
/>
`;
